openedx/core/djangoapps/theming/helpers.py

```diff
@@ -52,3 +71,311 @@ def get_themed_template_path(relative_path, default_path, **kwargs):
+    """
+    return True if settings.COMPREHENSIVE_THEME_DIR else False
+
+
+def get_site_theme_cache_key(site):
+    """
+    Return cache key for the given site.
+
+    Example:
+        >> site = Site(domain='red-theme.org', name='Red Theme')
+        >> get_site_theme_cache_key(site)
+        'theming.site.red-theme.org'
+
+    Parameters:
+        site (django.contrib.sites.models.Site): site where key needs to generated
+    Returns:
+        (str): a key to be used as cache key
+    """
+    cache_key = "theming.site.{domain}".format(
+        domain=site.domain
+    )
+    return cache_key
+
+
+def cache_site_theme_dir(site, theme_dir):
+    """
+    Cache site's theme directory.
+
+    Example:
+        >> site = Site(domain='red-theme.org', name='Red Theme')
+        >> cache_site_theme_dir(site, 'red-theme')
+
+    Parameters:
+        site (django.contrib.sites.models.Site): site for to cache
+        theme_dir (str): theme directory for the given site
+    """
+    cache.set(get_site_theme_cache_key(site), theme_dir, settings.THEME_CACHE_TIMEOUT)
+
+
+def get_static_file_url(asset):
+    """
+    Returns url of the themed asset if asset is not themed than returns the default asset url.
+
+    Example:
+        >> get_static_file_url('css/lms-main.css')
+        '/static/red-theme/css/lms-main.css'
+
+    Parameters:
+        asset (str): asset's path relative to the static files directory
+
+    Returns:
+        (str): static asset's url
+    """
+    return staticfiles_storage.url(asset)
+
+
+def get_themes():
+    """
+    get a list of all themes known to the system.
+    Returns:
+        list of themes known to the system.
+    """
+    themes_dir = get_base_theme_dir()
+    # pick only directories and discard files in themes directory
+    theme_names = []
+    if themes_dir:
+        theme_names = [_dir for _dir in os.listdir(themes_dir) if is_theme_dir(themes_dir / _dir)]
+
+    return [Theme(name, name) for name in theme_names]
+
+
+def is_theme_dir(_dir):
+    """
+    Returns true if given dir contains theme overrides.
+    A theme dir must have subdirectory 'lms' or 'cms' or both.
+
+    Args:
+        _dir: directory path to check for a theme
+
+    Returns:
+        Returns true if given dir is a theme directory.
+    """
+    theme_sub_directories = {'lms', 'cms'}
+    return bool(os.path.isdir(_dir) and theme_sub_directories.intersection(os.listdir(_dir)))
+
+
+class Theme(object):
+    """
+    class to encapsulate theme related information.
+    """
+    name = ''
+    theme_dir = ''
+    path = ''
+
+    def __init__(self, name='', theme_dir=''):
+        """
+        init method for Theme
+        Args:
+            name: name if the theme
+            theme_dir: directory name of the theme
+        """
+        self.name = name
+        self.theme_dir = theme_dir
+        self.path = Path(get_base_theme_dir()) / theme_dir / get_project_root_name()
```

Comment:
Should not theme paths be `self.path = Path(get_base_theme_dir()) / theme_dir`? if we want to get themes path with project root or component we can created a method in this class to return component path.